shared/actions/fs/index.js

```diff
@@ -15,22 +15,17 @@ import {isMobile} from '../../constants/platform'
 import {type TypedState} from '../../util/container'
 import {putActionIfOnPath, navigateAppend} from '../route-tree'
 
-function* listFavoritesSaga(): Saga.SagaGenerator<any, any> {
-  const state: TypedState = yield Saga.select()
-  try {
-    const results = yield Saga.call(RPCTypes.apiserverGetWithSessionRpcPromise, {
-      args: [{key: 'problems', value: '1'}],
-      endpoint: 'kbfs/favorite/list',
-    })
-    const username = state.config.username || ''
-    const loggedIn = state.config.loggedIn
-    const folders = Constants.folderToFavoriteItems(results && results.body, username, loggedIn)
-
-    yield Saga.put(FsGen.createFavoritesLoaded({folders}))
-  } catch (e) {
-    logger.warn('Error listing favorites:', e)
-  }
-}
+const loadFavorites = (state: TypedState) =>
```

Comment:
you likely want your error path also, else it'll black bar

Reply:
You don't  need the return type. Better to just let flow infer

Reply:
I’m OK with either way here, but I find having a return type on the function signature helpful for reading code sometimes especially on long or complicated functions. Is there a reason not to have them other than not having to figure the exact type and type then put?

Reply:
I think overly typing functions that are used within a module (where flows inference is strongest) usually ends up making things overly verbose. One of flows strengths is its inference capabilities. Sometimes the inferred type can be very complex and would actually be hard to type correctly. Obviously not here. If you feel like it aids in readability that's fine but adding the extra types can often just make the code more noisy

Reply:
@chrisnojima I'm pretty perplexed as to when we're supposed to type things explicitly. Earlier the guidance was that Flow sometimes defaults to `any`, which defeats a lot of the type checking we want to rely on. So I errred on the side of specifying type whenever possible. But it seems that now the guidance is that we should rely on inference when we can.

Is there a check we should be running to test this?

Reply:
When it's externally facing you have to type it. When it's just in the module, flow can totally infer the input and output so yo don't need typing. You can test this out. Note things like star work different inside vs out. (aka not at all outside a module). When in doubt flow coverage and introspect the type yourself with type-at-pos